silkrpc/http/request_handler.cpp

```diff
@@ -34,128 +36,185 @@
-                    reply.status = http::Reply::unauthorized;
-                    reply.headers.reserve(2);
-                    reply.headers.emplace_back(http::Header{"Content-Length", std::to_string(reply.content.size())});
-                    reply.headers.emplace_back(http::Header{"Content-Type", "application/json"});
-                    SILKRPC_INFO << "handle_request t=" << clock_time::since(start) << "ns\n";
-                    co_return;
-                }
-
-                // Validate token
-                auto verifier = jwt::verify()
-                    .allow_algorithm(jwt::algorithm::hs256{*jwt_secret_});
-
-                SILKRPC_TRACE << "jwt client token: " << client_token << " jwt_secret: " << *jwt_secret_ << "\n";
-                verifier.verify(decoded_client_token);
-            } catch (const std::system_error& se) {
-                SILKRPC_ERROR << "JWT invalid token: " << se.what() << "\n";
-                reply.content = make_json_error(request_id, 403, "invalid token").dump() + "\n";
-                reply.status = http::Reply::unauthorized;
-                reply.headers.reserve(2);
-                reply.headers.emplace_back(http::Header{"Content-Length", std::to_string(reply.content.size())});
-                reply.headers.emplace_back(http::Header{"Content-Type", "application/json"});
-                SILKRPC_INFO << "handle_request t=" << clock_time::since(start) << "ns\n";
-                co_return;
-            }
-        }
+    http::Reply reply;
+    if (request.content.empty()) {
+        reply.content = "";
+        reply.status = http::StatusType::no_content;
+    } else {
+        SILKRPC_DEBUG << "handle_request content: " << request.content << "\n";
 
         const auto request_json = nlohmann::json::parse(request.content);
-        request_id = request_json["id"].get<uint32_t>();
+        auto request_id = request_json["id"].get<uint32_t>();
         if (!request_json.contains("method")) {
             reply.content = make_json_error(request_id, -32600, "method missing").dump() + "\n";
-            reply.status = http::Reply::bad_request;
-            reply.headers.reserve(2);
-            reply.headers.emplace_back(http::Header{"Content-Length", std::to_string(reply.content.size())});
-            reply.headers.emplace_back(http::Header{"Content-Type", "application/json"});
-            SILKRPC_INFO << "handle_request t=" << clock_time::since(start) << "ns\n";
-            co_return;
+            reply.status = http::StatusType::bad_request;
+        } else {
+            const auto error = co_await is_request_authorized(request_id, request);
+            if (error.has_value()) {
+                reply.content = make_json_error(request_id, 403, error.value()).dump() + "\n";
+                reply.status = http::StatusType::unauthorized;
+            } else {
+                co_await handle_request(request_json, reply);
+            }
         }
+    }
```

Comment:
2 reasons:

1. before return we must call co_await do_write(reply); (line 65) and that for every exit condition
2. Just for logging purpose line 67 writes down the handling duration

It seems to me more readable in this way, instead of repeating 2 lines several times....